django/views/generic/edit.py

```diff
@@ -49,13 +48,10 @@ def get_form_kwargs(self):
 
     def get_success_url(self):
         """Return the URL to redirect to after processing a valid form."""
-        if self.success_url:
-            # Forcing possible reverse_lazy evaluation
-            url = force_text(self.success_url)
-        else:
-            raise ImproperlyConfigured(
-                "No URL to redirect to. Provide a success_url.")
-        return url
+        if not self.success_url:
+            raise ImproperlyConfigured("No URL to redirect to. Provide a success_url.")
+        # Forcing possible reverse_lazy evaluation
+        return str(self.success_url)
```

Comment:
I think `return str(self.success_url)  # success_url may be lazy` as you did elsewhere looks good.